lib/jxl/jxl_test.cc

```diff
@@ -1465,6 +1465,31 @@ TEST(JxlTest, JXL_TRANSCODE_JPEG_TEST(RoundtripJpegRecompressionToPixels420)) {
                                      /*distmap=*/nullptr, &pool));
 }
 
+TEST(JxlTest,
+     JXL_TRANSCODE_JPEG_TEST(RoundtripJpegRecompressionToPixels420EarlyFlush)) {
+  ThreadPoolInternal pool(8);
+  const PaddedBytes orig =
+      ReadTestData("imagecompression.info/flower_foveon.png.im_q85_420.jpg");
+  CodecInOut io;
+  io.dec_target = jxl::DecodeTarget::kQuantizedCoeffs;
+  ASSERT_TRUE(SetFromBytes(Span<const uint8_t>(orig), &io, &pool));
+
+  CodecInOut io2;
+  ASSERT_TRUE(SetFromBytes(Span<const uint8_t>(orig), &io2, &pool));
+
+  CompressParams cparams;
+  cparams.color_transform = jxl::ColorTransform::kYCbCr;
+
+  DecompressParams dparams;
+  dparams.max_downsampling = 8;
```

Comment:
If it's not too much trouble, it might be worthwhile to have this test in `decode_test.cc` instead (or in addition), and decode a recompressed jpeg while flushing at various random bitstream truncation points (not just after the DC like it is here), to check that nothing breaks in case there are e.g. a few AC groups already available too.

Reply:
AC groups don't use a specialized code path for the "flush" case though, so I don't think it's too useful...